### PR TITLE
packages yum: add MariaDB 10.4 build task

### PR DESCRIPTION
**Pull request by kenhys**
TODO:  need to fix some build error.
https://travis-ci.org/mroonga/mroonga/jobs/572162590

### CODE REVIEW

File: packages/yum/build-rpm.sh [hunk 1]
```diff
@@ -155,6 +155,24 @@ name = MariaDB
 baseurl = http://yum.mariadb.org/10.3/${distribution}${distribution_version}-${mariadb_architecture}
 gpgkey=https://yum.mariadb.org/RPM-GPG-KEY-MariaDB
 gpgcheck=1
+REPO
+        run yum install -y MariaDB-devel
+        if [ ${distribution_version} = 6 ]; then
+          run yum install -y cmake28
+        fi
+        ;;
+      mariadb-10.4-${PACKAGE})
```
Comment (kou) [inline packages/yum/build-rpm.sh]: Can we unify `mariadb-10.X-${PACKAGE}` clauses?

File: configure.ac [hunk 2]
```diff
@@ -209,9 +209,14 @@ AC_DEFUN([CONFIG_OPTION_MYSQL],[
 
   MYSQL_CXXFLAGS="-felide-constructors"
   case "$MYSQL_MAJOR_MINOR_VERSION" in
-  5.5|5.6|10.*)
+  5.5|5.6|10.[0-3])
     MYSQL_CXXFLAGS="$MYSQL_CXXFLAGS -fno-rtti"
     ;;
+  10.4)
```
Comment (kou) [inline configure.ac]: Could you put this before `5.5|5.6|...)`?

File: packages/rpm/centos/Makefile.am [hunk 3]
```diff
@@ -6,6 +6,7 @@ EXTRA_DIST =					\
 	mariadb-10.1-mroonga.spec.in		\
 	mariadb-10.2-mroonga.spec.in		\
 	mariadb-10.3-mroonga.spec.in		\
+	mariadb-10.4-mroonga.spec.in		\
```
Comment (kou) [inline packages/rpm/centos/Makefile.am]: This file is missing.

File: packages/yum/build-rpm.sh (hunk 1 above)
Comment (kenhys) [reply packages/yum/build-rpm.sh]: I've fixed them for simplicity.

File: configure.ac (hunk 2 above)
Comment (kenhys) [reply configure.ac]: I've fixed.

File: packages/rpm/centos/Makefile.am (hunk 3 above)
Comment (kenhys) [reply packages/rpm/centos/Makefile.am]: Oh...
I've added.

**Pull request merged**